packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx

```diff
@@ -291,6 +291,18 @@ export const ReleaseDetailsLayout = ({
     dispatch(releaseApi.util.invalidateTags([{ type: 'ReleaseAction', id: 'LIST' }]));
   };
 
+  const getCreatedByUser = () => {
+    if (!release?.createdBy) {
+      return null;
+    }
+
+    if (release.createdBy.lastname) {
+      return `${release.createdBy.firstname} ${release.createdBy.lastname}`;
+    }
+
+    return release.createdBy.firstname;
+  };
+
```

Comment:
Should we try to make this function more similar to the one from [content-manager](https://github.com/strapi/strapi/blob/f1dd5cc8eef574bac6679aab6f93276e57497328/packages/core/admin/admin/src/content-manager/utils/users.ts) ?

I mean return first the username, then firstname and lastname and finally email

Reply:
I think you should, it makes the product more consistent.

Reply:
Yup good idea!

Reply:
yeah I agree, regarding the rest of the code, to me it seems ok and also it works

Reply:
I updated to return username first if it it exists. I'm a little confused by the getDisplayName function, maybe I'm missing something but the logic doesn't seem right:

```tsx
  if (username) {
    return username;
  }

  // firstname is not required if the user is created with a username
  if (firstname) {
    return formatMessage(
      {
        id: 'global.fullname',
        defaultMessage: '{firstname} {lastname}',
      },
      {
        firstname,
        lastname,
      }
    ).trim();
  }

  return email;
```

Given a user MUST have either a username or a firstname

- If we have a username we return early, but we also check for a firstname, but why? If do not have a username then we should at least have a firstname? If we do not have a firstname then we should have a username for which we already returned early.
- If that's correct then `return email` will never happen since we returned for either username or firstname
- Anyway here for content-releases we sanitize out the `email` via

https://github.com/strapi/strapi/blob/779adab77adeae5df22a37d783f9368a09f6c044/packages/core/content-releases/server/src/controllers/release.ts#L79

I would be surprised if it isn't done the same over there 🤔 but I didn't check

Reply:
When a user is registered with SSO the only thing we can be sure is there is email, for that reason is the fallback. Then a user could have username (is not mandatory so we check that first). And in the most common case user should have firstname (mandatory in the admin level, but as mentioned with SSO you could have an empty firstname)

About the sanitize, I think you're right. We don't need to use the permissionsManager here because the only sensible info on the Release is the user, so I guess is okay to sanitize only it directly